src/parser/demonhunter/havoc/modules/talents/Demonic.tsx

```diff
@@ -42,12 +45,13 @@ class Demonic extends Analyzer {
     this.addEventListener(Events.cast.by(SELECTED_PLAYER).spell(SPELLS.DEATH_SWEEP), this.onDeathSweepCast);
   }
 
-  onEyeBeamCast(event) {
+  onEyeBeamCast(event: CastEvent) {
     const hasMetaBuff = this.selectedCombatant.hasBuff(SPELLS.METAMORPHOSIS_HAVOC_BUFF.id, event.timestamp - 1000);
 
-    if (hasMetaBuff) {
+    if (hasMetaBuff || !(this.selectedCombatant.hasTalent(SPELLS.TRAIL_OF_RUIN_TALENT) || this.selectedCombatant.hasTalent(SPELLS.FIRST_BLOOD_TALENT))) {
       return;
     }
+
     this.eyeBeamCasts += 1;
     this.eyeBeamTimeStamp = event.timestamp;
 
```

Comment:
I can't comment on the line but im pretty sure that deathsweepsInMetaCounter will never be undefined now meaning this is dead code